src/zenml/steps/utils.py

```diff
@@ -335,6 +346,10 @@ def resolve_output_artifact(
             artifact: A TFX artifact type.
             data: The object to be passed to `handle_return()`.
         """
+        # Make an exception for BaseArtifact, return artifact
```

Comment:
Either the comment or the implementation is wrong here, because it actually returns None

Reply:
Fixed